tilecloud_chain/generate.py

```diff
@@ -176,21 +189,34 @@ def add_elapsed_togenerate(metatile):
             gene.add_metatile_splitter()
             gene.imap(Logger(logger, logging.INFO, '%(tilecoord)s'))
 
-            if gene.layer['type'] != 'mapnik' or gene.layer['output_format'] != 'grid':
+            if self.count_tiles is None:
```

Comment:
Because that's ultimately what the condition was checking­, except that checking if the counter is initialised is less brittle than checking if the conditions that would have led to the counter being initialised are false (especially in cases like slave mode where there is no layer in gene)